### PR TITLE
fix: Avoid creation of duplicate sessions when using favorited search query

### PR DESCRIPTION
**Pull request by adam-wolfe**
## Proposed changes

This fix addresses [Issue 1029](https://github.com/zowe/vscode-extension-for-zowe/issues/1029) in which executing a favorited search query creates a duplicate session in the Data Sets and USS trees. 

This PR resolves the issue with datasets by removing a call to labelRefresh() in packages/zowe-explorer/src/dataset/DatasetTree.ts, which adds a space to the label text of the tree item causing a second tree item to be created when a favorited search query is executed after running a search via the text entry bar.

For favorited USS searches, the sanitizedPath has been moved from the label to the description of the tree item.

## Release Notes

Milestone: 2.6.1

Changelog:

- Removed call to labelRefresh() in packages/zowe-explorer/src/dataset/DatasetTree.ts
- Moved the sanitizedPath display for USS from the tree item label to the description for USS search queries
- Added function expandSession to ZoweTreeProvider.ts which should reliably expand session nodes
- Prevented favorited search nodes from becoming expandable in the dataset view
  - This may be undesirable. TBD

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/dataset/DatasetTree.ts
Comment (adam-wolfe) [inline packages/zowe-explorer/src/dataset/DatasetTree.ts:885]: I did not experience any issues with this function call removed.

**Review by traeok — COMMENTED** [review]
I noticed that the session node does not expand after searching for a query within this node, but we might be able to change the state of the session node to expand after the refresh.

Otherwise, LGTM! I will approve once there are more perspectives regarding the preferred behavior.

**Review by t1m0thyj — COMMENTED** [review]
Thanks for the fix 🙂 Observed some issues while testing this PR. Not sure whether they are related - I'm happy to file them as separate issues if you prefer.

**USS Tree**
1. Favorite a USS search query.
2. Restart VS Code or reload the window.
3. Recall the favorited query and notice the node is labelled correctly.
4. Expand and collapse the search node to reset its status.
5. Recall the favorited query again and notice the node is labelled incorrectly.

ze-uss-favorites

**Data Sets Tree**
1. Favorite a data set search query.
2. Restart VS Code or reload the window.
3. Try to recall the favorited query and notice it doesn't work.

ze-ds-favorites

**Comment by adam-wolfe** [comment]
I'll take a look at the USS issue, the label must be getting modified somewhere. 

I see the same dataset behavior in the v2.6.0 Zowe release. Trying to expand the favorited search query results in error popups.

**Comment by adam-wolfe** [comment]
Added expandSession function to ZoweTreeProvider.ts. This commit should resolve the issue with the nodes failing to expand after searching in both the USS and the Dataset views.

**Comment by adam-wolfe** [comment]
Still seeing issues with the USS view. Not ready for review.

**Comment by adam-wolfe** [comment]
The nodes should expand reliably after searching now. I also made it so the favorited dataset queries no longer become expandable after executing a query. Attempting to expand a search query in the dataset view would produce error messages, so I think it makes sense to not enable expansion for favorited dataset queries.

File: packages/zowe-explorer/src/abstract/ZoweTreeProvider.ts
Comment (traeok) [inline packages/zowe-explorer/src/abstract/ZoweTreeProvider.ts:270]: I did some quick testing and noticed that this fix will still work without the reveal call using `expand: false` (line 270).

**Review by traeok — APPROVED** [review]
LGTM! Did some testing with the tree views and everything is working as intended. Thanks @adam-wolfe !

File: packages/zowe-explorer/src/abstract/ZoweTreeProvider.ts
Comment (adam-wolfe) [reply packages/zowe-explorer/src/abstract/ZoweTreeProvider.ts:270]: With that line commented out, there are expansion issues in the USS view:
expansion1

**Review by zFernand0 — APPROVED** [review]

**Review by zFernand0 — CHANGES_REQUESTED** [review]
just waiting on the changelog merge conflicts 😋

File: packages/zowe-explorer/CHANGELOG.md
Comment (traeok) [inline packages/zowe-explorer/CHANGELOG.md:36]: Line 36 should be above `2.6.0`, but I accidentally put it in the `2.6.0` section for my previous PR - sorry about that! It should be up where your changelog comment is.

**Review by traeok — DISMISSED** [review]
LGTM! Thanks @adam-wolfe

File: packages/zowe-explorer/CHANGELOG.md
Comment (adam-wolfe) [reply packages/zowe-explorer/CHANGELOG.md]: Should be resolved now

**Review by traeok — APPROVED** [review]
Appreciate the fix for the changelog. Was going to add it to my other maintenance PR but you beat me to it 😅

**Review by zFernand0 — APPROVED** [review]
Thanks for fixing the changelog!
LGTM! 😋

**Comment by zFernand0** [comment]
merging this PR due to the following reasons:
- It has the required number of approvals
- the theia tests failures are transient
- the audit will be fixed in: https://github.com/zowe/vscode-extension-for-zowe/pull/2106

**Pull request merged**
